packages/broker/src/plugins/legacyWebsocket/WebsocketPlugin.ts

```diff
@@ -43,12 +45,11 @@ export class WebsocketPlugin extends Plugin<WebsocketPluginConfig> {
         this.websocketServer = new WebsocketServer(
             httpServer,
             this.networkNode,
-            new StreamFetcher(this.brokerConfig.streamrUrl),
+            new StreamFetcher(this.getRestUrl()),
             this.publisher,
-            this.metricsContext,
+            (await (this.streamrClient!.getNode())).getMetricsContext(),
```

Comment:
I noticed this same line before
```ts
(await (this.client.getNode())).getMetricsContext()
```
and it seems a bit cumbersome to get the metrics context. I'm wondering whether it would be possible to have the `metricsContext` already "injected" into the plugin base class instance as a field before `start`?

Reply:
True, added that improvement to "Future improvements" list